src/Chapter3.hs

```diff
@@ -971,6 +1076,31 @@ implement the following functions:
 🕯 HINT: to implement this task, derive some standard typeclasses
 -}
 
+data WeekDay = Sunday 
+    | Monday 
+    | Tuesday 
+    | Wednesday
+    | Thursday 
+    | Friday 
+    | Saturday
+    deriving (Show, Enum)
+
+isWeekend :: WeekDay -> Bool
+isWeekend day = case day of
+    Saturday -> True
+    Sunday -> True
+    _ -> False
+
+nextDay :: WeekDay -> WeekDay
+nextDay day = case day of
+    Saturday -> Sunday      -- Is there a way in which I can avoid a special case for "Saturday" ??
+    _ -> succ day
+
+daysToParty :: WeekDay -> Int
+daysToParty day = case day of 
+    Saturday -> (length $ enumFromTo day Friday) + 1
```

Comment:
You can use the `fromEnum` method of the `Enum` typeclass to implement this function in a slightly more efficient way :slightly_smiling_face: